Add MAINTAINERS.adoc (#103)
Vandit1604/jenkins-docs: MAINTAINERS.adoc

= How to Version Documentation

== 1. Create Versioned Branch:

> - Begin by creating a dedicated branch for a new release, following semantic versioning, for example, _2.426.1_.

Reviewer: ```suggestion
- Begin by creating a dedicated branch for a new release, following semantic versioning, for example, _2.426.x_.
```

== 2. Update `antora.yml` in Versioned Branches:

- Ensure that all `antora.yml` files within this branch contain a consistent value for the `version` key. This value will be displayed to the users accessing the documentation.

== 3. Add Branch to `antora-playbook.yml`:

> - Add the newly created branch, such as _**2.426.1**_, into the `antora-playbook.yml` file located in the playbook directory. This inclusion instructs Antora to fetch this branch and generate documentation for it, aligned with the version specified in the `antora.yml` file.

Reviewer: ```suggestion
- Add the newly created branch, such as _**2.426.x**_, into the `antora-playbook.yml` file located in the playbook directory. This inclusion instructs Antora to fetch this branch and generate documentation for it, aligned with the version specified in the `antora.yml` file.
```

[source,yaml]
----
content:
  sources:
    - url: https://github.com/PATH/jenkins-docs.git
      branches: [main,2.426.1]
      start_paths: [docs/tutorials, docs/user-docs, docs/solutions]
      # developer docs are un-versioned that's why they are to be fetched individually
    - url: https://github.com/Vandit1604/jenkins-docs.git
      branches: [main,2.426.1]
      start_paths: [docs/dev-docs, docs/events, docs/security, docs/sigs, docs/projects, docs/images, docs/books, docs/community, docs/project, docs/about, docs/download]
----

> - The `content.sources.branches` key should be updated with the name of the newly created _**2.426.1**_ branch

Reviewer: ```suggestion
- The `content.sources.branches` key should be updated with the name of the newly created _**2.426.x**_ branch
```

= How to Version Documentation

== 1. Create Versioned Branch:

- Begin by creating a dedicated branch for a new release, following semantic versioning, for example, _2.426.x_ since software release aren't mapped to patch releases but to LTS stable releases.

> - This branch should be created 2 weeks prior the minor release when the release candidate is released. 

Reviewer: ```suggestion
```

> NOTE: The main branch will contain documentation for the next release and when a release candidate will be released and another branch is created from the `main` branch. In simple terms, The `main` branch will always contains documentation for next release and 2 weeks before the release, we will create a branch. 

Reviewer: ```suggestion
NOTE: The main branch will contain documentation for the next release and when a release candidate will be released and another branch is created from the `main` branch. In simple terms, The `main` branch will always contains documentation for the next release and some time before the release, we will create a versioned branch. 
```
